monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/ScanObservable.scala

```diff
@@ -42,22 +42,31 @@ class ScanObservable[A,R](
           implicit val scheduler = out.scheduler
           private[this] var isDone = false
           private[this] var state = initialState
+          // Signal initial state immediately
+          private[this] var lastAck = out.onNext(state)
```

Comment:
You don't need to implement back-pressure on each `onNext`. This shouldn't be a `var`, because it's only the first `onNext` that's out of place:

```scala
private[this] val firstAck = out.onNext(state)
```